src/tools/inner_node.rs

```diff
@@ -26,6 +27,29 @@ pub struct InnerNode {
     pub crypto: Arc<Crypto>,
     pub tls: Tls,
     pub ident: String,
+    pub handshake_errors: Option<HandshakeErrors>,
```

Comment:
We already have one `TestConfig` so I placed these bools right in InnerNode. We could probably refactor this a bit and placed then inside something like `HandshakeData` together with `ident` field. Maybe we'll do it later?

Reply:
> Maybe we'll do it later?

Sure, no probs, I think it's even better to do it in a separate commit

> then when we see the config.test.something keyword used in other parts of code, our eyes quickly skip it

But yeah, what I meant with this comment is everything that is doing something funny/unexpected compared to the normal synth node behaviour could be encapsulated in something which will clearly state it's used only for testing purposes.
Because when I see this code snippet in the `handshake.rs` without knowing any extra context:
```rust
if self.handshake_flip_bit_pub_key {
    randomly_flip_bit(&mut public_key) {
}
```
First, I think it is like "_why on earth are we flipping bits here_"

But if I see this:
```rust
if self.test.handshake_flip_bit_pub_key {
    randomly_flip_bit(&mut public_key) {
}
```
My thinking process is like: "_okay, some test stuff is happening here, ignore_mode=on..."_